tests/api2/test_smb_groupmap.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+
+from middlewared.test.integration.utils import call
+from middlewared.test.integration.assets.account import group
+
+BASE_RID_GROUP = 200000
+
+
+@pytest.mark.parametrize('groupname,expected_memberof,expected_rid', [
+    ('builtin_administrators', 'S-1-5-32-544', 512),
+    ('builtin_guests', 'S-1-5-32-546', 514)
+])
+def test__local_builtin_accounts(groupname, expected_memberof, expected_rid):
+    entry = call('group.query', [['group', '=', groupname]], {'get': True})
+    rid = int(entry['sid'].split('-')[-1])
+    assert rid == expected_rid
+
+    groupmap = call('smb.groupmap_list')
+    assert str(entry['gid']) in groupmap['local_builtins']
+    assert groupmap['local_builtins'][str(entry['gid'])]['sid'] == entry['sid']
+
+    members = call('smb.groupmap_listmem', expected_memberof)
+    assert entry['sid'] in members
+
+
+def test__local_builtin_users_account():
+    entry = call('group.query', [['group', '=', 'builtin_users']], {'get': True})
+
+    rid = int(entry['sid'].split('-')[-1])
+    assert rid == entry['id'] + BASE_RID_GROUP
+
+    members_dom_users = call('smb.groupmap_listmem', 'S-1-5-32-545')
+    assert entry['sid'] in members_dom_users
+
+
+def test__new_group():
+    with group({"name": "group1"}) as g:
+        # Validate GID is being assigned as expected
+        assert g['sid'] is not None
+        rid = int(g['sid'].split('-')[-1])
+        assert rid == g['id'] + BASE_RID_GROUP
```

Comment:
I don't _think_ I saw a **user** test similar to the above (where the `sid` & `rid` are checked).  Maybe add one?

Reply:
Yeah, that will be next major PR to improve passdb handling.